kong/runloop/balancer.lua

```diff
@@ -961,11 +961,19 @@ local function post_health(upstream, hostname, ip, port, is_healthy)
     return nil, "no healthchecker found for " .. tostring(upstream.name)
   end
 
+  local ok, err
   if ip then
-    return balancer:setAddressStatus(is_healthy, ip, port, hostname)
+    ok, err = healthchecker:set_target_status(ip or hostname, port, hostname, is_healthy)
```

Comment:
`ip` is always not `nil` here.

```suggestion
    ok, err = healthchecker:set_target_status(ip, port, hostname, is_healthy)
```

Reply:
@locao Fixed, thank you!